### ISSUE TITLE
Property in list doesn't like to be in kebab-case

### ISSUE THREAD

**Issue by chkpnt** [micronaut-projects/micronaut-core#3424]
I'm trying to use a type-safe `@ConfigurationProperty` for a configuration like

```yaml
example:
  some-kebab-property: yufka
  nested:
    - simple: simple-0
      some-other-kebab: yufka-0
      foobar: foobar-0
    - simple: simple-1
      some-other-kebab: yufka-1
      foobar: foobar-1
    - simple: simple-2
      some-other-kebab: yufka-2
      foobar: foobar-2
```

I'm using such a configuration class:

```java
@ConfigurationProperties("example")
public class ExampleConfiguration {

    private String someKebabProperty;
    private List<NestedConfiguration> nested;

    // getter & setter

    public static final class NestedConfiguration {

        private String simple;
        private String someOtherKebab;
        private String foobar;

        // getter & setter
    }
}
```

This works fine but for the property `some-other-kebab` within the list. Actually, I'm unsure if this is a bug or if I've just forget some annotations.

### Task List

- [x] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Checkout the example project
2. Execute the test

### Expected Behaviour

The test should be green.

### Actual Behaviour

The last assert fails:

<img width="1303" alt="grafik" src="https://user-images.githubusercontent.com/1956979/83903080-d07ef600-a75d-11ea-9307-828ddc5b73e1.png">

### Environment Information

- **Operating System**: macOS
- **Micronaut Version:** 1.3.5
- **JDK Version:** 1.8

### Example Application

[@chkpnt/micronaut-playground](https://github.com/chkpnt/micronaut-playground)

**Comment by jameskleeh** [comment]
This was resolved with https://github.com/micronaut-projects/micronaut-core/pull/3209. Will be released in 1.3.6 soon

**Comment by chkpnt** [comment]
Whoa.... that "soon" has been really very soon!

I've just updated to 1.3.6 and can confirm: It's working well :-)

<img width="946" alt="grafik" src="https://user-images.githubusercontent.com/1956979/83924515-29ae5000-a785-11ea-895f-451dedb1806d.png">
